app/com/linkedin/drelephant/mapreduce/heuristics/DistributedCacheLimitHeuristic.java

```diff
@@ -0,0 +1,148 @@
+  private static final String MAPREDUCE_JOB_CACHE_ARCHIVES_FILESIZES = "mapreduce.job.cache.archives.filesizes";
+  private static final String MAPREDUCE_JOB_CACHE_FILES = "mapreduce.job.cache.files";
+  private static final String MAPREDUCE_JOB_CACHE_ARCHIVES = "mapreduce.job.cache.archives";
+  private static long distributedCacheFileSizeLimit = 500000000; // 500MB default
+  private HeuristicConfigurationData _heuristicConfData;
+
+  public DistributedCacheLimitHeuristic(HeuristicConfigurationData heuristicConfData) {
+    this._heuristicConfData = heuristicConfData;
+    loadParameters();
+  }
+
+  private void loadParameters() {
+    Map<String, String> paramMap = _heuristicConfData.getParamMap();
+    String heuristicName = _heuristicConfData.getHeuristicName();
+
+    String cacheLimit = paramMap.get(DISTRIBUTED_CACHE_FILE_SIZE_LIMIT_CONF);
+    if (cacheLimit != null) {
+      try {
+        distributedCacheFileSizeLimit = Long.parseLong(cacheLimit);
+        logger.info(
+            heuristicName + " will use " + DISTRIBUTED_CACHE_FILE_SIZE_LIMIT_CONF + " with the following setting: "
+                + distributedCacheFileSizeLimit);
+      } catch (NumberFormatException e) {
+        logger
+            .warn("Error parsing " + DISTRIBUTED_CACHE_FILE_SIZE_LIMIT_CONF + " from the conf file. Check for typos...",
+                e);
+      }
+    }
+  }
+
+  @Override
+  public HeuristicResult apply(MapReduceApplicationData data) {
+    if (data == null || !data.getSucceeded()) {
+      return null;
+    }
+    
+    String jobName = data.getJobName();
+    Properties jobConf = data.getConf();
+    String cacheFiles = jobConf.getProperty(MAPREDUCE_JOB_CACHE_FILES, null);
+    String cacheFileSizes = jobConf.getProperty(MAPREDUCE_JOB_CACHE_FILES_FILESIZES, null);
+
+    HeuristicResult result = null;
+
+    if (cacheFiles != null && cacheFileSizes != null) {
+      result =
+          new HeuristicResult(_heuristicConfData.getClassName(), _heuristicConfData.getHeuristicName(), Severity.NONE,
+              0);
+      String[] cacheFilesArray = cacheFiles.split(",");
+      String[] cacheFileSizesArray = cacheFileSizes.split(",");
+
+      if (cacheFilesArray.length != cacheFileSizesArray.length) {
+        result.setSeverity(Severity.MODERATE);
+        logger.warn("Mismatch in the number of files and their corresponding sizes for " + MAPREDUCE_JOB_CACHE_FILES);
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_FILES, Integer.toString(cacheFilesArray.length));
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_FILES_FILESIZES, Integer.toString(cacheFileSizesArray.length));
+        return result;
+      }
+
+      Map<String, String> cacheFileToSizeMap = new HashMap<String, String>();
+      for (int i = 0; i < cacheFilesArray.length; i++) {
+        cacheFileToSizeMap.put(cacheFilesArray[i], cacheFileSizesArray[i]);
+      }
+
+      if (checkFileSizeLimit(result, cacheFileToSizeMap)) {
```

Comment:
You are checking if files and archives are separately > 500 MB. I think the combined size should not exceed 500 MB is the intent. Can you get that clarified from jack ?  Also, some comments what the difference between the two would be good.